src/pybind/mgr/cephadm/template.py

```diff
@@ -0,0 +1,44 @@
+import copy
+import logging
+from typing import Optional
+
+from jinja2 import Environment, PackageLoader, select_autoescape
+
+
+logger = logging.getLogger(__name__)
+
+
+class TemplateEngine:
+    def render(self, name: str, context: Optional[dict] = None) -> str:
+        raise NotImplementedError()
+
+
+class Jinja2Engine(TemplateEngine):
+    def __init__(self):
+        self.env = Environment(
+            loader=PackageLoader('cephadm', 'templates'),
+            autoescape=select_autoescape(['html', 'xml']),
+            trim_blocks=True,
+            lstrip_blocks=True
+        )
+
+    def render(self, name: str, context: Optional[dict] = None) -> str:
+        template = self.env.get_template(name)
+        if context is None:
+            return template.render()
+        return template.render(context)
+
+
+class TemplateMgr:
+    def __init__(self):
+        self.engine = Jinja2Engine()
+        self.managed_header_variable = 'cephadm_managed'
+        self.managed_header_text = 'This file is generated by cephadm.'
```

Comment:
```suggestion
        self.base_context = {
            'cephadm_managed' = 'This file is generated by cephadm.',
        } 
```
?

Reply:
Fixed, the user can also override the value now.